### PR TITLE
NODE-2348: warn on deprecated event and options

### PR DESCRIPTION
**Pull request by mbroadst**
When using the unified topology, a number of options and events are now deprecated. This will notify users of that, and point them to our documentation, which now contains information about deprecations.

### CODE REVIEW

File: lib/core/sdam/topology.js [hunk 1]
```diff
@@ -110,6 +117,14 @@ class Topology extends EventEmitter {
     }
 
     options = Object.assign({}, common.TOPOLOGY_DEFAULTS, options);
+    DEPRECATED_OPTIONS.forEach(optionName => {
+      if (options[optionName]) {
+        process.emitWarning(
```
Comment (daprahamian) [inline lib/core/sdam/topology.js]: `process.emitWarning` is not supported in node 4. You should probably export [this function](https://github.com/mongodb/node-mongodb-native/blob/master/lib/utils.js#L581) and use that instead.

File: lib/operations/connect.js [hunk 2]
```diff
@@ -438,6 +438,18 @@ function createServer(mongoClient, options, callback) {
   });
 }
 
+const DEPRECATED_UNIFIED_EVENTS = new Set(['reconnect', 'reconnectFailed', 'attemptReconnect']);
+function registerDeprecatedEventNotifiers(client) {
+  client.on('newListener', eventName => {
+    if (DEPRECATED_UNIFIED_EVENTS.has(eventName)) {
+      process.emitWarning(
```
Comment (daprahamian) [inline lib/operations/connect.js]: same as above.

File: lib/core/sdam/topology.js (hunk 1 above)
Comment (mbroadst) [reply lib/core/sdam/topology.js]: done

File: lib/operations/connect.js (hunk 2 above)
Comment (mbroadst) [reply lib/operations/connect.js]: done

**Pull request merged**